wiki/Ranking_Criteria/osu!mania/en.md

```diff
@@ -1,78 +1,202 @@
-| 10K                                                    | MuziBox, Pump It Up (Double)                                                    |
-| 12K                                                    | Keyboardmania (Light)                                                           |
-| 14K                                                    |                                                                                 |
-| 16K                                                    | beatmaniaIIDX (Double)                                                          |
-| 18K                                                    | E5 - Dancing Super Station (Double).                                            |
-| 20K                                                    |                                                                                 |
-
-Not included: Keyboardmania (Light+, Real) \[24K\] and (Double) \[48K\].
+- **Column**: Vertical lines on the playfield where notes are placed. The editor uses `0|1|2|3|4|5|6|7|8|9` for timestamp labels. 
+- **Key mode**: The number of columns present in a difficulty.
+- **Key sounding**: Hit sounding using samples extremely similar or directly from the song in their respective places. This method of hit sounding usually provides low feedback to the player.
+- **Slider velocity**: The speed at which the notes/bar lines move on the playfield. Base slider velocity can be controlled in the timing panel and additional changes can be made through inherited (green) timing points.
+- **Long-term slider velocity**: A slider velocity change lasting more than 4 measures.
+- **Musical phrase**: A standalone unit of musical meter that is usually 4 measures long. 
+- **Short-term slider velocity**: A slider velocity change lasting less than 1/2 of a beat.
+- **Stutter**: Repeated alternations between two slider velocity values with a noticeable difference in their values.
+- **Teleport**: A high slider velocity that causes a note to move at a speed that is hard or impossible to react to.
+
+### Patterns
+
+- **Anchor**: A pattern that has a relatively high amount of notes in one column compared with another column, usually found within different types of streams.
+- **Chord**: Two or more notes pressed at the same time.
+- **Chordstream (Jumpstream, Handstream, Quadstream)**: Streams that mix different types of chords. 
+- **Grace**: Two or more notes pressed in rapid succession, generally using snaps at or above 1/6.
+- **Hand**: Three notes pressed at the same time.
+- **Inverse**: A type of pattern that replaces all regular notes with long notes in a given section, commonly when there are enough long notes to cover a majority of the playfield.
+- **Jack**: Three or more consecutive notes in the same column, usually pressed at a faster pace than surrounding notes.
+- **Jump**: Two notes pressed at the same time.
+- **Jumptrill**: A trill consisting of two alternating jumps.
```

Comment:
idk if jump trill and jumptrill are two different things but the draft later on mentions jump trill which is confusing since jump and trill are defined so is the thing later a different thing or the same or two things at once